src/mavlink-router/mainloop.cpp

```diff
@@ -794,10 +826,15 @@ void Mainloop::_handle_pipe()
           }
 
           std::unique_ptr<UdpEndpoint> udp{new UdpEndpoint()};
-          if (udp->open(address.c_str(), port, a[5]=="1") < 0) {
+          if (udp->open(address.c_str(), port, eavesdropping) < 0) {
               log_error("Could not open %s:%d", address.c_str(), port);
               continue;
           }
+          udp->set_coalescing(coalesce_num_bytes, coalesce_timeout);
+          for (int id : nodelay_ids) {
+              udp->add_message_to_nodelay(id);
+          }
+
           _add_dynamic_endpoint(name, command, udp.release());
```

Comment:
making this testable in unit test is somewhat straight:
- make this function to add (and remove) dynamic endpoint public
- that BTW is the reason why I (and every other coding standard out there) STRONGLY discourage encoding things like "protected/private" in method names using _ ... this creates absolutely useless work
- call add_dynamic_endpoint and verify in the same way as before
- completely independently verify that sample commands sent to the pipe are parsed correctly